lesson-15/src/index.js

```diff
@@ -0,0 +1,140 @@
+(async () => {
+  async function getCategories() {
+    const response = await fetch('https://dummyjson.com/products/categories')
+    const categories = await response.json()
+    return categories
+  }
+
+  console.log(await getCategories())
+
+  const listOfCategories = await getCategories()
+
+  console.log(listOfCategories.length)
+
+  function findCategoryByName(arrayOfCategories, nameOfCategory) {
+    return arrayOfCategories.find((category) => category.name === nameOfCategory)
+  }
+
+  console.log(findCategoryByName(listOfCategories, 'Groceries'))
+  console.log(findCategoryByName(listOfCategories, 'Electronics'))
+
+  // 4
+  async function getCategoryProducts(arrayOfCategories, nameOfCategory) {
+    const category = arrayOfCategories.find((categoryItem) => categoryItem.name === nameOfCategory)
+    if (category !== undefined) {
```

Comment:
```suggestion
    if (category) {
```

проверка if вернет false, если category имеет значение undefined или другие “ложные” значения (null, 0, ‘’, false) - явно сравнивать не нужно

Reply:
Я когда это читаю, все понимаю, но когда делаю напрочь забываю(

Reply:
@yzubr с практикой, когда пишешь такие проверки почти каждый день, запоминается быстрее - у тебя всё ещё впереди)